src/main/java/frc/robot/subsystems/tank/TankSubsystem.java

```diff
@@ -44,18 +60,51 @@ public TankSubsystem() {
     rightFollow.setInverted(InvertType.FollowMaster);
     rightFollow.setNeutralMode(NeutralMode.Brake);
 
+    // Initialize Talon sensors
+    // https://docs.ctre-phoenix.com/en/latest/ch14_MCSensor.html#cross-the-road-electronics-magnetic-encoder-absolute-and-relative
+    // https://docs.ctre-phoenix.com/en/latest/ch14_MCSensor.html#software-select-sensor
+    leftMain.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+    rightMain.configSelectedFeedbackSensor(FeedbackDevice.CTRE_MagEncoder_Relative, 0, 0);
+
+    // https://docs.ctre-phoenix.com/en/latest/ch14_MCSensor.html#sensor-phase
+    leftMain.setSensorPhase(false);
+    rightMain.setSensorPhase(false);
+
+    // Initialize navX AHRS
+    // https://www.kauailabs.com/public_files/navx-mxp/apidocs/java/com/kauailabs/navx/frc/AHRS.html
+    ahrs = new AHRS(SPI.Port.kMXP); 
+
+    // Initialize odometry class
+    // https://docs.wpilib.org/en/stable/docs/software/advanced-controls/state-space/state-space-pose_state-estimators.html
+    odometry = new DifferentialDrivePoseEstimator(new Rotation2d(), new Pose2d(),
+      new MatBuilder<>(Nat.N5(), Nat.N1()).fill(0.02, 0.02, 0.01, 0.02, 0.02), // State measurement standard deviations. X, Y, theta.
+      new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.02, 0.02, 0.01), // Local measurement standard deviations. Left encoder, right encoder, gyro.
+      new MatBuilder<>(Nat.N3(), Nat.N1()).fill(0.1, 0.1, 0.01)); // Global measurement standard deviations. X, Y, and theta. 
+
     // Initialize power values
     leftPower = 0;
     rightPower = 0;
+
+    zeroPosition();
   }
 
   @Override
   public void periodic() {
-    // This method will be called once per scheduler run
-    // TODO: odometry
 
     leftMain.set(ControlMode.PercentOutput, leftPower);
     rightMain.set(ControlMode.PercentOutput, rightPower);
+
+    // Update odometry readings
+    Rotation2d gyroAngle = Rotation2d.fromDegrees(ahrs.getAngle());
+    DifferentialDriveWheelSpeeds wheelVelocities = new DifferentialDriveWheelSpeeds(
+      leftMain.getSelectedSensorVelocity() * ENCODER_TICKS_TO_INCHES, 
+      rightMain.getSelectedSensorVelocity() * ENCODER_TICKS_TO_INCHES);
+    double leftDistance = leftMain.getSelectedSensorPosition() * ENCODER_TICKS_TO_INCHES;
+    double rightDistance = rightMain.getSelectedSensorPosition() * ENCODER_TICKS_TO_INCHES;
+
+    odometry.update(gyroAngle, wheelVelocities, leftDistance, rightDistance);
```

Comment:
Seems like this interface isn't great. We're passing in a `DifferentialDriveWheelSpeeds` object and then separate positions for left and right wheels. These two interfaces should match. I know this isn't your interface, but keep this in mind.